*** frescobaldi_app/lilydoc/network.py.orig	2013-08-31 15:19:42.000000000 +0200
--- frescobaldi_app/lilydoc/network.py	2013-08-31 15:17:40.000000000 +0200
*************** def langs():
*** 60,66 ****
          lang = s.value("language", "", type(""))
          if not lang:
              try:
!                 lang = locale.getdefaultlocale()[0]
              except ValueError:
                  return []
          if not lang or lang == "none":
--- 60,67 ----
          lang = s.value("language", "", type(""))
          if not lang:
              try:
!                 locale.setlocale(locale.LC_ALL, '')
!                 lang = locale.getlocale()[0]
              except ValueError:
                  return []
          if not lang or lang == "none":
