*** frescobaldi_app/hyphendialog.py.orig	2013-09-05 15:05:02.000000000 +0200
--- frescobaldi_app/hyphendialog.py	2013-09-05 15:07:38.000000000 +0200
*************** import widgets
*** 39,44 ****
--- 39,45 ----
  import hyphdicts
  import hyphenator
  
+ from po import defaultLanguageFromQLocale
  
  # paths to check for hyphen dicts
  default_paths = [
*************** class HyphenDialog(QDialog):
*** 126,139 ****
              lastused = settings().value("lastused", "", type(""))
              if lastused:
                  yield lastused
-             try:
-                 lang = locale.getdefaultlocale()[0]
-             except ValueError:
-                 pass
              else:
!                 yield lang
!                 yield lang.split('_')[0]
!                 
          langs = [item[1] for item in self._langs]
          for preselect in select():
              try:
--- 127,141 ----
              lastused = settings().value("lastused", "", type(""))
              if lastused:
                  yield lastused
              else:
!                 try:
!                     lang = defaultLanguageFromQLocale() or locale.getdefaultlocale()[0]
!                 except ValueError:
!                     pass
!                 if lang:
!                     yield lang
!                     yield lang.split('_')[0]
!         
          langs = [item[1] for item in self._langs]
          for preselect in select():
              try:
