*** frescobaldi_app/language_names/__init__.py.orig	2013-08-31 15:19:42.000000000 +0200
--- frescobaldi_app/language_names/__init__.py	2013-08-31 15:17:40.000000000 +0200
*************** def languageName(code, language=None):
*** 33,39 ****
      
      """
      if language is None:
!         language = locale.getdefaultlocale()[0] or "C"
          
      for lang in (language, language.split('_')[0], "C"):
          try:
--- 33,40 ----
      
      """
      if language is None:
!         locale.setlocale(locale.LC_ALL, '')
!         language = locale.getlocale()[0] or "C"
          
      for lang in (language, language.split('_')[0], "C"):
          try:
