*** frescobaldi_app/mainwindow.py.orig	2013-08-31 17:06:18.000000000 +0200
--- frescobaldi_app/mainwindow.py	2013-08-31 17:05:42.000000000 +0200
*************** class ActionCollection(actioncollection.
*** 987,992 ****
--- 987,997 ----
          
          self.help_manual.setShortcuts(QKeySequence.HelpContents)
          
+         # roles
+         self.file_quit.setMenuRole(QAction.QuitRole)
+         self.edit_preferences.setMenuRole(QAction.PreferencesRole)
+         self.help_about.setMenuRole(QAction.AboutRole)
+         
      def translateUI(self):
          self.file_new.setText(_("action: new document", "&New"))
          self.file_open.setText(_("&Open..."))
