*** frescobaldi_app/scorewiz/__init__.py.orig	2013-08-31 14:40:41.000000000 +0200
--- frescobaldi_app/scorewiz/__init__.py	2013-08-31 14:17:01.000000000 +0200
*************** class Actions(actioncollection.ActionCol
*** 61,66 ****
--- 61,67 ----
          self.scorewiz.setShortcut(QKeySequence("Ctrl+Shift+N"))
          self.newwithwiz = QAction(parent)
          self.newwithwiz.setIcon(icons.get("tools-score-wizard"))
+         self.scorewiz.setMenuRole(QAction.NoRole)
          
      def translateUI(self):
          self.scorewiz.setText(_("Setup New Score (in current document)..."))
