*** frescobaldi_app/po/setup.py.orig	2013-08-31 15:19:42.000000000 +0200
--- frescobaldi_app/po/setup.py	2013-08-31 15:17:40.000000000 +0200
*************** def current():
*** 44,50 ****
      language = QSettings().value("language", "", type(""))
      if not language:
          try:
!             language = locale.getdefaultlocale()[0]
          except ValueError:
              pass
      if not language:
--- 44,51 ----
      language = QSettings().value("language", "", type(""))
      if not language:
          try:
!             locale.setlocale(locale.LC_ALL, '')
!             language = locale.getlocale()[0]
          except ValueError:
              pass
      if not language:
